docs/index.rst

```diff
@@ -20,6 +20,7 @@
     architecture/infrastructure.rst
     architecture/application.rst
     migrating.rst
+    upgrade-marshmallow.rst
```

Comment:
after reviewing and checking with the documentation I think the better place for it would be Tutorials/Marshmallow 3 Compatibility